### ISSUE TITLE
Permission issue when rebuilding

### ISSUE THREAD

**Issue by Rukkulakki** [microsoft/navcontainerhelper#2210]
PLEASE DO NOT INCLUDE ANY PASSWORDS OR TOKENS IN YOUR ISSUE!!!

**Describe the issue**
When trying to remake a container the script fails as it doesn't have the proper permissions.
I've been developing using an 18.2 version of BC. Now we wanted to make sure that everything still works on the 19.1 OnPrem version, so I wanted to remove to old container and create a new.

**Scripts used to create container and cause the issue**
```
$containerName = 'lanitol'
$password = 'P@ssw0rd'
$securePassword = ConvertTo-SecureString -String $password -AsPlainText -Force
$credential = New-Object pscredential 'admin', $securePassword
$auth = 'UserPassword'
$artifactUrl = Get-BcArtifactUrl -type 'OnPrem' -version '19.1' -country 'dk' -select 'Latest'
$licenseFile = 'd:\nav\infar\udvikling\bc19.flf'
New-BcContainer `
    -accept_eula `
    -containerName $containerName `
    -credential $credential `
    -auth $auth `
    -artifactUrl $artifactUrl `
    -imageName 'bc19cu1' `
    -licenseFile $licenseFile `
    -shortcuts DesktopFolder `
    -updateHosts
```

**Full output of scripts**
```
BcContainerHelper is version 2.0.22
BcContainerHelper is running as administrator
Host is Microsoft Windows 11 Pro - 21H2
Docker Client Version is 20.10.10
Docker Server Version is 20.10.10
Removing container lanitol
Removing lanitol from container hosts file
Removing lanitol-* from container hosts file
Remove-Item : Adgang til cloudfilen blev nægtet
At C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\2.0.22\Misc\Remove-DesktopShortcut.ps1:26 char:21
+                     Remove-Item $folderName -Recurse -force
+                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Remove-Item], Win32Exception
    + FullyQualifiedErrorId : System.ComponentModel.Win32Exception,Microsoft.PowerShell.Commands.RemoveItemCommand...
```

**Additional context**
The problem is the Desktop folder, or the files within. For some reason the files now need administrator rights to delete. And even though I run powershell as Admin the script fails. 
As soon as I manually delete the folder, everything is fine.

**Comment by freddydk** [comment]
I am puzzled by the "access to the cloud-file was denied"
Is your desktop setup with something special?

**Comment by Rukkulakki** [comment]
Define special :) 
OneDrive synchronizes Pictures, Documents, Teams-chatfiles and Desktop. But other than that there is nothing touching the files.

**Comment by freddydk** [comment]
Does it work if you place the shortcuts on the desktop (not in a folder)?
I will try to setup onedrive sync of my desktop.

**Comment by Rukkulakki** [comment]
Seems to be only if the shortcuts are inside the folder. 
Using -shortcuts Desktop allows for deleting them when recreating the container; regardless of shortcuts setting. There however I get an error from OneDrive asking me to check if I had enough permissions for the shortcuts to by synchronised. The script and I are allowed to delete the shortcuts, so it is somehow OneDrive trying to "take ownership" of the files that fails.
Manually deleting the shortcuts from the folder also lets the script recreate the container. Here it ignores the folder when deleting the old setup, and just reuses the folder when creating the shortcuts.
Using CommonDesktopFolder allows the script to delete shortcuts and folder when recreating the container.

**Comment by freddydk** [comment]
I have added some resilience to the code in order to avoid failing when removing the folder (which is the problem)

**Comment by freddydk** [comment]
shipped in 2.0.23
